BinaryStudioAcademy/bsa-2023-squirrel
# 55: Setup SignalR connection from Console App to WebApi (upd)

This PR instead [previous](https://github.com/BinaryStudioAcademy/bsa-2023-squirrel/pull/59)
• Changed Console App flow and refactored some modules
• Added SignalR connections at current version of ConsoleApp
• Configured hub for using all available actions with User DB
• Added and checked Installer from tast [#83](https://github.com/BinaryStudioAcademy/bsa-2023-squirrel/pull/75) (work manually and from UI)
• Registered and configured SignalR Hub on SQL Web Api side


**How to check:**
1. Build all projects
2. Download and install Squirrel Console App (....\backend\Squirrel.ConsoleApp.Setup\Debug\setup.exe)
3. Run SQL Web Api project
4. Run Squirrel Console App
5. Set Db connections settings
ex: 
```
POST http://localhost:44567/setting/connect
{
  "DbType": "SqlServer",
  "ConnectionString": "Server=localhost;Database=SquirrelCoreDB;Trusted_Connection=True;"
}
```
In response you will get your ClientId like **bf06fe9b-077b-4910-a994-faf37e0db6df**

![image](https://github.com/BinaryStudioAcademy/bsa-2023-squirrel/assets/107005256/54753e4d-fcea-4c5b-bdf9-f762b7d7e891)

6. Now, you can send requests for actions from SQL Web Api Swagger 
          Pass the ClientId
          For some actions you can set additional filters
```
{
  "clientId": "bf06fe9b-077b-4910-a994-faf37e0db6df",
  "filterName": "string", // optional
  "filterRowsCount": 0 // optional
}

```
![image](https://github.com/BinaryStudioAcademy/bsa-2023-squirrel/assets/107005256/1f10010e-fd2b-4ffb-936a-3bcf029d7f97)

In that moment SQL Web Api execute method at Console App
Then Console App will call the processing received data method at SQL Web Api

7. If all are OK - you will see text representation of response at Console window (SQL Web Api)

![image](https://github.com/BinaryStudioAcademy/bsa-2023-squirrel/assets/107005256/dd0806c8-9381-4ea6-a3ae-ac59fff3dde1)
